mlir/lib/Bindings/Python/IRTypes.cpp

```diff
@@ -618,6 +618,15 @@ class PyMemRefType : public PyConcreteType<PyMemRefType, PyShapedType> {
               return mlirMemRefTypeGetLayout(self);
             },
             "The layout of the MemRef type.")
+        .def_property_readonly(
+            "strides_and_offset",
+            [](PyMemRefType &self) -> std::pair<std::vector<int64_t>, int64_t> {
+              std::vector<int64_t> strides(mlirShapedTypeGetRank(self));
```

Comment:
MemRef is ranked right? ShapedType has a private method [https://github.com/llvm/llvm-project/blob/main/mlir/lib/Bindings/Python/IRTypes.cpp#L447](requireHasRank) that I thought about making public but then realized it's probably superfluous because MemRef must have been verified to be ranked at somet point (should double check though).

Reply:
Ah I misread the type. SG